# 📓 00 · Carga inicial de datos

**Objetivo del notebook:** cargar el dataset de cepas de cannabis y darle un primer vistazo para
saber "con qué estamos trabajando".

---
### 🧠 Un poco de teoría antes de empezar

- **DataFrame:** es la estructura central de pandas. Piénsalo como una **hoja de cálculo** dentro de Python:
  filas (cada cepa) y columnas (cada característica).
- **¿Por qué cargamos desde un CSV local y no desde internet?** El dataset original vive en Hugging Face,
  pero la librería para descargarlo (`datasets`) está rota en Python 3.13. Además, cargar desde un archivo
  local es **más rápido y reproducible**: cualquiera que clone el repo obtiene exactamente los mismos datos.

El código de carga vive en `src/data_loading.py` para mantener este notebook limpio.

## 1. Importar herramientas y cargar los datos

In [1]:
import sys
sys.path.append("..")   # permite que Python "vea" la carpeta src/ (que está un nivel arriba)

import pandas as pd
from src.data_loading import load_raw

# Mostrar todas las columnas cuando imprimamos el DataFrame (por defecto oculta algunas)
pd.set_option("display.max_columns", 40)

df = load_raw()
print(f"Filas (cepas):     {df.shape[0]}")
print(f"Columnas (rasgos): {df.shape[1]}")

Filas (cepas):     8910
Columnas (rasgos): 38


## 2. Primer vistazo: las primeras filas

`df.head()` muestra las primeras 5 filas. Es el primer gesto de todo análisis: ver la forma real de los
datos antes de sacar conclusiones.

In [2]:
df.head()

,strain_name,breeder,description,current_price_gbp,original_price_gbp,discount_percent,pack_options,product_url,overview,growth_and_harvest,experience,seed_type,flowering_period_type,indica_sativa,type_ratio,strain_type_summary,environment,strength,medical_strains,smell_taste,effect,climate,flavor,thc,cbd,yield_indoor,yield_outdoor,height_indoor,height_outdoor,indoor_height_detail,indoor_flowering_time,outdoor_harvest_time,flowering_time,harvest_month,genetic_background,stock_availability,sale_item,most_popular_seeds
0,Sour Diesel,Royal Queen Seeds,The world famous Sour Diesel is now available ...,18.54,20.60,10.0,3 Feminised Seeds (GBP 18.54) | 5 Feminised Se...,https://www.seed-city.com/components/com_virtu...,"This is Sour Diesel, a high-quality cannabis s...",This strain grows well in both indoor and outd...,Users report a balanced and enjoyable experien...,Feminized,12/12 Photoperiod,Sativa Dominant,30% Indica / 70% Sativa,Sativa Dominant,"Indoor, Outdoor",Normal,"Appetite Stimulation, Nausea, Anxiety","Chemical, Herbal, Sour","Relaxed, Energetic",Mediterranean,Citrus,21.8%,1.5%,406.0g/m²,523.0g/plant,166.0cm,152.0cm,Medium (81cm - 140cm),Medium (56 to 90 days),The End of October,64 days,Mid October,Northern Lights x Haze,In Stock,Cannabis Seeds Currently on Sale,Most Popular Cannabis Seeds
1,Big Bud Auto,Seed Stockers,"Legendary strain with huge buds, easy growth, ...",15.30,17.00,10.0,3 Feminised Seeds (GBP 15.30) | 5 Feminised Se...,https://www.seed-city.com/components/com_virtu...,"This is Big Bud Auto, a high-quality cannabis ...",This strain grows well in both indoor and outd...,Users report a balanced and enjoyable experien...,Feminized,Autoflowering,50% Indica/50% Sativa,50% Indica / 50% Sativa,50% Indica/50% Sativa,"Indoor, Outdoor, Greenhouse",Super Strength,"Insomnia, Pain Relief",Sweet,"Energetic, Relaxed",Tropical,"Sweet, Citrus, Pine",20.9%,1.0%,581.0g/m²,905.0g/plant,171.0cm,151.0cm,Medium (81cm - 140cm),Medium (56 to 90 days),Middle of October,90 days,Early October,Colombian x Mexican,In Stock,Cannabis Seeds Currently on Sale,Most Popular Cannabis Seeds
2,Pineapple Express,FastBuds,High-potency strain with tropical flavors and ...,24.12,26.80,10.0,3 Feminised Seeds (GBP 24.12) | 5 Feminised Se...,https://www.seed-city.com/components/com_virtu...,"This is Pineapple Express, a high-quality cann...",This strain grows well in both indoor and outd...,Users report a balanced and enjoyable experien...,Feminized,Autoflowering,50% Indica/50% Sativa,50% Indica / 50% Sativa,50% Indica/50% Sativa,"Indoor, Outdoor",Super Strength,"Appetite Stimulant, Social","Fruity, Pineapple, Sweet","Euphoric, Focused",Temperate,"Earthy, Pine",24.2%,1.9%,514.0g/m²,1052.0g/plant,148.0cm,248.0cm,Medium (81cm - 140cm),Medium (56 to 90 days),Middle of October,60 days,September,Afghani x Mexican,In Stock,Cannabis Seeds Currently on Sale,Most Popular Cannabis Seeds
3,Strawberry Cough,Dutch Passion Seeds,Potent Sativa seeds with fruity flavor and upl...,23.68,29.60,20.0,3 Feminised Seeds (GBP 23.68) | 5 Feminised Se...,https://www.seed-city.com/components/com_virtu...,"This is Strawberry Cough, a high-quality canna...",This strain grows well in both indoor and outd...,Users report a balanced and enjoyable experien...,Feminized,12/12 Photoperiod,Sativa Dominant,30% Indica / 70% Sativa,Sativa Dominant,Indoor,Normal,Euphoric,Sweet,Creative,Continental,"Berry, Earthy, Pine",20.4%,1.6%,606.0g/m²,663.0g/plant,125.0cm,174.0cm,Medium (81cm - 140cm),Medium (56 to 90 days),The End of October,67 days,Early October,Afghani x Thai,In Stock,Cannabis Seeds Currently on Sale,Most Popular Cannabis Seeds
4,White Runtz,Anesia Seeds,"Popular California strain with sweet, fruity t...",31.40,28.08,40.0,3 Feminised Seeds (GBP 31.40) | 5 Feminised Se...,https://www.seed-city.com/components/com_virtu...,"This is White Runtz, a high-quality cannabis s...",This strain grows well in both indoor and outd...,Users report a balanced and enjoyable experien...,Feminized,12

### Otras formas de mirar los datos

No basta con ver el principio. Conviene mirar también el **final** y algunas filas **al azar**, por si los
datos cambian de forma a lo largo del archivo (algo común en datos scrapeados).

`df.tail()` muestra las **últimas** filas:

In [3]:
df.tail()

,strain_name,breeder,description,current_price_gbp,original_price_gbp,discount_percent,pack_options,product_url,overview,growth_and_harvest,experience,seed_type,flowering_period_type,indica_sativa,type_ratio,strain_type_summary,environment,strength,medical_strains,smell_taste,effect,climate,flavor,thc,cbd,yield_indoor,yield_outdoor,height_indoor,height_outdoor,indoor_height_detail,indoor_flowering_time,outdoor_harvest_time,flowering_time,harvest_month,genetic_background,stock_availability,sale_item,most_popular_seeds
8905,Chutney and Cheese,Ultra Genetics,High-yielding indica strain with mouth-waterin...,2.00,28.08,40.0,1 Single Feminised Seed (GBP 2.99) | 100 Femin...,https://www.seed-city.com/images/out-of-stock-...,"Chutney and Cheese Ultra Genetics is a new, in...",This strain grows well in both indoor and outd...,Users report a balanced and enjoyable experien...,Feminized,12/12 Photoperiod,Indica Dominant,70% Indica / 30% Sativa,Indica Dominant,"Indoor, Outdoor",Normal,Relaxing,"Cheese, Fruity, Sweet",Relaxing,Mediterranean,Citrus,20%,Low,450 - 500 gr/m2,500 - 550 gr/plant,80 - 120 cm,140 - 180 cm,Medium (81cm - 140cm),Medium (56 to 90 days),Middle of October,8-9 Weeks,October,Northern Lights x Haze,Out of Stock,Cannabis Seeds Currently on Sale,Most Popular Cannabis Seeds
8906,Auto Animal Zookies,Advanced Seeds,Potent hybrid strain with complex flavor profi...,889.64,28.08,40.0,3 Feminised Seeds (GBP 889.64) | 5 Feminised S...,https://www.seed-city.com/components/com_virtu...,"This is White Widow, a high-quality cannabis s...",This strain grows well in both indoor and outd...,Users report a balanced and enjoyable experien...,Feminized,12/12 Photoperiod,Indica Dominant,70% Indica / 30% Sativa,Indica Dominant,"Indoor, Outdoor",Normal,Relaxing,Sweet,Relaxing,Mediterranean,Citrus,20%,Low,450 - 500 gr/m2,500 - 550 gr/plant,80 - 120 cm,140 - 180 cm,Medium (81cm - 140cm),Medium (56 to 90 days),Middle of October,8-9 Weeks,October,Northern Lights x Haze,In Stock,Cannabis Seeds Currently on Sale,Most Popular Cannabis Seeds
8907,Blue Dream,Barneys Farm Seeds,"West Coast strain with high THC, fruity flavor...",1.00,28.08,40.0,3 Feminised Seeds (GBP 152.30) | 5 Feminised S...,https://www.seed-city.com/components/com_virtu...,"Blue Dream, a sativa-dominant hybrid high in T...",This strain grows well in both indoor and outd...,Users report a balanced and enjoyable experien...,"Regular , Feminized",12/12 Photoperiod,Sativa Dominant,70% Indica / 30% Sativa,Indica Dominant,"Indoor, Outdoor",Normal,"Depression, Euphoric, General Pain, Happy, Rel...","Berry, Sweet",Relaxing,Mediterranean,Citrus,High - 16% - 20%,Low - 0.1% - 1%,450 - 500 gr/m2,500 - 550 gr/plant,80 - 120 cm,140 - 180 cm,Tall (140cm plus),Medium (56 to 90 days),Middle of October,8-9 Weeks,October,Northern Lights x Haze,In Stock,Cannabis Seeds Currently on Sale,Most Popular Cannabis Seeds
8908,Bruce Banner,Barneys Farm Seeds,Potent strain with 60% Sativa and 40% Indica. ...,1.00,28.08,40.0,3 Feminised Seeds (GBP 152.30) | 5 Feminised S...,https://www.seed-city.com/components/com_virtu...,Blackskull's Bruce Banner Strain is notably po...,This strain grows well in both indoor and outd...,Users report a balanced and enjoyable experien...,Feminized,12/12 Photoperiod,Sativa Dominant,70% Indica / 30% Sativa,Indica Dominant,"Indoor, Outdoor",Super Strength,"Anxiety, Depression, General Pain, Insomnia, N...","Citrus, Sweet",Intense euphoric head high with increased ener...,Mediterranean,Citrus,20%,Low,450 - 500 gr/m2,500 - 550 gr/plant,80 - 120 cm,140 - 180 cm,Medium (81cm - 140cm),Medium (56 to 90 days),Beginning of October,63-70 days,early October,Northern Lights x Haze,In Stock,Cannabis Seeds Currently on Sale,Most Popular Cannabis Seeds
8909,Blueberry,Nirvana Seeds,Indica strain with relaxing effects. Cross of ...,626.99,28.08,40.0,3 Feminised Seeds (GBP 626.99) | 5 Feminised S...,https://www.seed-city.com/components/com_virtu...,"This is White Widow, a high-quality cannabis s...",

`df.sample(3)` toma 3 filas **al azar**. Es útil para no dejarnos engañar por el orden del archivo
(cada vez que lo ejecutes saldrán filas distintas):

In [4]:
df.sample(3)

,strain_name,breeder,description,current_price_gbp,original_price_gbp,discount_percent,pack_options,product_url,overview,growth_and_harvest,experience,seed_type,flowering_period_type,indica_sativa,type_ratio,strain_type_summary,environment,strength,medical_strains,smell_taste,effect,climate,flavor,thc,cbd,yield_indoor,yield_outdoor,height_indoor,height_outdoor,indoor_height_detail,indoor_flowering_time,outdoor_harvest_time,flowering_time,harvest_month,genetic_background,stock_availability,sale_item,most_popular_seeds
6215,Chem Toffees,Holy Smoke Seeds,"Indica strain with dark choc, coffee, caramel,...",6.50,28.08,40.0,1 Single Regular Seed (GBP 6.50),https://www.seed-city.com/images/discontinued-...,"This is White Widow, a high-quality cannabis s...",This strain grows well in both indoor and outd...,Users report a balanced and enjoyable experien...,"Regular , Feminized",12/12 Photoperiod,Indica Dominant,70% Indica / 30% Sativa,Indica Dominant,Indoor,Normal,"Body Stone, Insomnia, Relaxing",Caramel,Relaxing,Mediterranean,Citrus,20%,Low,450 - 500 gr/m2,500 - 550 gr/plant,80 - 120 cm,140 - 180 cm,Medium (81cm - 140cm),Medium (56 to 90 days),Middle of October,8-9 Weeks,October,Northern Lights x Haze,Out of Stock,Cannabis Seeds Currently on Sale,Most Popular Cannabis Seeds
429,Auto Critical Kush,Barneys Farm Seeds,"Fast-growing, compact plant with heavy colas. ...",6.92,28.08,40.0,1 Single Feminised Seed (GBP 6.92) | 3 Feminis...,https://www.seed-city.com/components/com_virtu...,"Meet Critical Kush Auto, the autoflowering ver...",This strain grows well in both indoor and outd...,Users report a balanced and enjoyable experien...,Feminized,Autoflowering,Indica Dominant,Feminised,Indica Dominant,"Indoor, Outdoor",Normal,"Cerebral, Insomnia, Relaxing","Earthy, Fruity, Piney, Woodsy","Relaxed, Calming, Sleepy, Cerebral",Continental,"Citrus, Sweet, Spicy",23.4%,1.2%,654.0g/m²,875.0g/plant,105.0cm,177.0cm,Short (up to 80cm),Medium (56 to 90 days),Middle of October,81 days,November,Afghani x Thai,In Stock,Cannabis Seeds Currently on Sale,Most Popular Cannabis Seeds
3147,Apricot Auto,FastBuds,"Up to 26% THC, 550 g/m2 yield in 9-10 weeks. I...",32.20,28.08,40.0,3 Feminised Seeds (GBP 32.20) | 5 Feminised Se...,https://www.seed-city.com/components/com_virtu...,"Apricot Auto, an Indica-dominant autoflower, s...",This strain grows well in both indoor and outd...,Users report a balanced and enjoyable experien...,Feminized,Autoflowering,Indica Dominant,70% Indica / 30% Sativa,Indica Dominant,"Indoor, Outdoor",Super Strength,"Appetite Stimulant, Body Stone, Couch Lock, Ha...","Citrus, Fruity, Sour, Sweet, Tangy, Woodsy",Relaxing,Mediterranean,Citrus,Up to 26%,< 1%,450 - 500 gr/m2,500 - 550 gr/plant,80 - 120 cm,140 - 180 cm,Medium (81cm - 140cm),Medium (56 to 90 days),Middle of October,8-9 Weeks,October,Northern Lights x Haze,In Stock,Cannabis Seeds Currently on Sale,Most Popular Cannabis Seeds


También podemos mirar **solo las columnas que nos interesan**, poniendo sus nombres entre corchetes.
Por ejemplo, el nombre de la cepa junto con su productor (breeder):

In [5]:
df[["strain_name", "breeder"]].head(10)

,strain_name,breeder
0,Sour Diesel,Royal Queen Seeds
1,Big Bud Auto,Seed Stockers
2,Pineapple Express,FastBuds
3,Strawberry Cough,Dutch Passion Seeds
4,White Runtz,Anesia Seeds
5,O.G.Kush,Seed Stockers
6,G13 Haze,Barneys Farm Seeds
7,LSD-25,FastBuds
8,Gelato,Blackskull Seeds
9,Golden Tiger,Ace Seed bank


## 3. ¿Qué columnas tenemos?

38 columnas es bastante. Veamos sus nombres. La explicación de cada una está en
[`docs/DICCIONARIO_DATOS.md`](../docs/DICCIONARIO_DATOS.md), agrupadas en 3 dimensiones:
**química/efectos**, **comercial/precios** y **botánica/cultivo**.

In [6]:
df.columns.tolist()

['strain_name',
 'breeder',
 'description',
 'current_price_gbp',
 'original_price_gbp',
 'discount_percent',
 'pack_options',
 'product_url',
 'overview',
 'growth_and_harvest',
 'experience',
 'seed_type',
 'flowering_period_type',
 'indica_sativa',
 'type_ratio',
 'strain_type_summary',
 'environment',
 'strength',
 'medical_strains',
 'smell_taste',
 'effect',
 'climate',
 'flavor',
 'thc',
 'cbd',
 'yield_indoor',
 'yield_outdoor',
 'height_indoor',
 'height_outdoor',
 'indoor_height_detail',
 'indoor_flowering_time',
 'outdoor_harvest_time',
 'flowering_time',
 'harvest_month',
 'genetic_background',
 'stock_availability',
 'sale_item',
 'most_popular_seeds']

### 📝 Qué aprendimos

- Tenemos **8.910 cepas** y **38 columnas**.
- Las primeras, últimas y aleatorias filas se ven consistentes: mismo formato en todo el archivo.
- A simple vista ya se notan cosas raras: `thc` y `cbd` aparecen como texto (`'21.8%'`), y columnas como
  `effect` traen **varias etiquetas separadas por comas**.

Todo eso lo investigamos a fondo en el siguiente notebook, **`01_exploracion`**.